torch/csrc/cuda/nccl.cpp

```diff
@@ -850,7 +855,12 @@ void all2all_single_unequal_split(
   for (const auto r : c10::irange(numranks)) {
     // NCCL uses 0 byte message for synchronization
     // Avoid send/recv when message size is zero
+#if defined(NCCL_MAJOR) && \
+    ((NCCL_MAJOR > 2) || ((NCCL_MAJOR == 2) && (NCCL_MINOR > 13)))
+    if (true) {
+#else
```

Comment:
Consider creating helper function
```
#if defined(NCCL_MAJOR) && \
    ((NCCL_MAJOR > 2) || ((NCCL_MAJOR == 2) && (NCCL_MINOR > 13)))
template<typename T>
constexpr bool _nccl_should_send(C10_UNUSED T _unused_) {
  return foo;
}
#else
template<typename T>
inline bool _nccl_should_send(C10_UNUSED T value) {
  return value != 0;
}
#endif 
```

or just `constexpr bool _nccl_supports_zero_send` and `(_nccl_supports_zero_send || original_cond)` it throughout the code